File: TME.Store/TME.Api.Domain/Models/ApiPrice.cs
namespace TME.Api.Domain.Models
{
    public class ApiPrice
    {
        public int Amount { get; set; }
        public float PriceValue { get; set; }
        public bool Special { get; set; }

        public ApiPrice(int amount, float priceValue, bool special)
        {
            this.Amount = amount;
            this.PriceValue = priceValue;
            this.Special = special;
        }
    }
}

File: tme_store/lib/api/models/api_price.dart
import 'package:json_annotation/json_annotation.dart';

part 'api_price.g.dart';

@JsonSerializable()
class ApiPrice {
  @JsonKey(name: 'Amount')
  final int amount;

  @JsonKey(name: 'PriceValue')
  final double value;

  @JsonKey(name: 'Special')
  final bool isSpecial;

  ApiPrice({
    this.amount,
    this.value,
    this.isSpecial
  });

  static ApiPrice fromJson(Map<String, dynamic> json) => _$ApiPriceFromJson(json);

  Map<String, dynamic> toJson() => _$ApiPriceToJson(this);
}